### DISCUSSION TITLE
custom onChange in useController

### DISCUSSION THREAD

**Discussion by deepgolani4** [react-hook-form/react-hook-form#9144] [General] [answered]
```
const { field:field1 } = useController({
    control,
    name: 'field1',
  });

<input {...field1}  onchange={()=>{}} />
```

Is there any way to overwrite onChange so I don't have to write custom onChange in the input field?
i can just pass the custom onChange from spreading the useController.

**Comment by bluebill1049** [comment, marked as answer]
```tsx
<input onChange={(e) => {
  field.onChange(e)
  // do your own change event
}} value={field.value} />
```

**Reply by RSchneider94** [reply]
hey @bluebill1049, I'll take the opportunity here as I've found this answer while I was looking for an answer to my problem (and even though I went waaay long looking in the other pages I haven't found it). My problem goes a bit further than this one but starts here 😛 .. so, this is the way to get a custom onChange, that's okay and works fine.. but what I have here on my side is: I have to take an action based on which checkbox the user has chosen and then update the form based on that. So I have a form that looks like this:

![image](https://user-images.githubusercontent.com/11929659/193948735-490e90ae-beb6-4f86-9711-fccf9df97e8d.png)
 
Then, whenever the user selects, e.g. "Manage" in case it has "View" selected before, I need to deselect "View" because with "Manage" user already has "View". I think you know what I mean. I have tried to make use of `useEffect` + `watch` + `setValue` but as the react-hooks linter warns about the need to add them as deps, you know what happens, infinite loop. So I'm wondering how it would be possible to achieve what I need. Because as the docs for the `onChange` inside `Controller` says:

> This prop update [formState](https://react-hook-form.com/api/useform/formstate) and you should avoid manually invoke [setValue](https://react-hook-form.com/api/useform/setvalue) or other API related to field update

Any help would be very welcome!

Thanks!

**Reply by leapful** [reply]
@RSchneider94 : Could you please create a new discussion for your problem with a codesandbox url ? Since your problem is way more complex than the question of this author topic, continue discussing only make more confusion for the author.

**Reply by RSchneider94** [reply]
> @RSchneider94 : Could you please create a new discussion for your problem with a codesandbox url ? Since your problem is way more complex than the question of this author topic, continue discussing only make more confusion for the author.

@leapful thanks! I have created a new discussion [here](https://github.com/react-hook-form/react-hook-form/discussions/9151).
